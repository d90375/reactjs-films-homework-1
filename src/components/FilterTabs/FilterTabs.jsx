import React from 'react';
import PropTypes from 'prop-types';
import FilterTab from '../FilterTab';

import styles from './FilterTabs.scss';

const FilterTabs = (props) => {
  const { genres, fetchByFilter, condition } = props;

  const defaultValue = Number(condition) ? condition : 'Genre';

  const className = Number(condition) ? `${styles.select} ${styles.active}` : styles.select;

  let options;

  if (genres) {
    options = genres.map((genre, index) => (
      <option
        key={genre}
        value={index}
      >
        {genre}
      </option>
    ));
  }

  return (
    <div className={styles.container}>
      <FilterTab fetchByFilter={fetchByFilter} condition={condition} filter="Trending">
        Trending
      </FilterTab>
      <FilterTab fetchByFilter={fetchByFilter} condition={condition} filter="Top Rated">
        Top Rated
      </FilterTab>
      <FilterTab fetchByFilter={fetchByFilter} condition={condition} filter="Coming soon">
        Coming soon
      </FilterTab>
      <select
        className={className}
        name="Genre"
        defaultValue={defaultValue}
        onChange={(e) => fetchByFilter(e.target.value)}
      >
        <option value="Genre" disabled hidden>Genre</option>
        {options}
      </select>
    </div>
  );
};

FilterTabs.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  fetchByFilter: PropTypes.func.isRequired,
  condition: PropTypes.string.isRequired,
};

export default FilterTabs;
