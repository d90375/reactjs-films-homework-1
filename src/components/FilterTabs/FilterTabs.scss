@import '../../mixins.scss';

.container {
  display: flex;
  flex-wrap: wrap;
}

.select {
  padding: 0 10px;
  width: 160px;
  font-weight: bold;
  color: #0aaee4;
  background-color: transparent;
  border: none;
  cursor: pointer;

  @include outline-none();
}

.active {
  border-bottom: 2px solid #0aaee4;
}
