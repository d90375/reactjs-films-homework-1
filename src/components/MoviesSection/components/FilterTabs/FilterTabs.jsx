import React from 'react';
import PropTypes from 'prop-types';
import FilterTab from '../FilterTab';

import styles from './FilterTabs.scss';

const FilterTabs = ({ genres, fetchByFilter }) => {
  let options;

  if (genres) {
    options = genres.map((genre) => <option key={genre}>{genre}</option>);
  }

  return (
    <div className={styles.container}>
      <FilterTab className={styles.active} fetchByFilter={fetchByFilter}>
        Trending
      </FilterTab>
      <FilterTab className={styles.active} fetchByFilter={fetchByFilter}>
        Top Rated
      </FilterTab>
      <FilterTab className={styles.active} fetchByFilter={fetchByFilter}>
        Coming soon
      </FilterTab>
      <select className={styles.select} defaultValue="Genre" onChange={(e) => fetchByFilter(e.target.value)}>
        <option value="Genre" disabled hidden>Genre</option>
        {options}
      </select>
    </div>
  );
};

FilterTabs.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  fetchByFilter: PropTypes.func.isRequired,
};

export default FilterTabs;
