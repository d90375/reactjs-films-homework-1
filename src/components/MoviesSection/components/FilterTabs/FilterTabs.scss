@import '../../../../mixins.scss';

.container {
  display: flex;
  flex-wrap: wrap;
}

.select {
  padding: 10px 10px 0;
  width: 160px;
  font-weight: bold;
  color: #0aaee4;
  background-color: transparent;
  border: none;
  cursor: pointer;

  @include outline-none();
}

.active {
  border-bottom: 2px solid #0aaee4;
}
