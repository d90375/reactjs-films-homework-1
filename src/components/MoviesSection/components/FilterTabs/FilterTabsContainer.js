import { connect } from 'react-redux';
import { setMoviesCondition } from '../../../../modules/movies/moviesAction';
import FilterTabs from './FilterTabs';

const mapDispatchToProps = {
  setMoviesCondition,
};

export default connect(
  null,
  mapDispatchToProps,
)(FilterTabs);
