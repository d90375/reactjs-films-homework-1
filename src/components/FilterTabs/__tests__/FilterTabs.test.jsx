import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create } from 'react-test-renderer';
import FilterTabs from '../FilterTabs';

describe('FilterTabs tests', () => {
  describe('FilterTabs render', () => {
    it('FilterTabs renders correctly when genres tab active', () => {
      const mockCallBack = jest.fn();
      const genres = ['drama', 'crime', 'detective'];
      const tree = create(
        <Router>
          <FilterTabs genres={genres} fetchByFilter={mockCallBack} condition="123" />
        </Router>,
      );
      expect(tree).toMatchSnapshot();
    });

    it('FilterTabs renders correctly when trending tab active', () => {
      const mockCallBack = jest.fn();
      const genres = ['drama', 'crime', 'detective'];
      const tree = create(
        <Router>
          <FilterTabs genres={genres} fetchByFilter={mockCallBack} condition="Trending" />
        </Router>,
      );
      expect(tree).toMatchSnapshot();
    });

    it('FilterTabs renders correctly when genres are null', () => {
      const mockCallBack = jest.fn();
      const genres = null;
      const tree = create(
        <Router>
          <FilterTabs genres={genres} fetchByFilter={mockCallBack} condition="Trending" />
        </Router>,
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe('onChange function', () => {
    it('onChange callback function called', () => {
      const mockCallBack = jest.fn();
      const genres = ['drama', 'crime', 'detective'];

      const tree = create(
        <Router>
          <FilterTabs
            genres={genres}
            fetchByFilter={mockCallBack}
            condition="123"
          />
        </Router>,
      );

      const select = tree.root.findByProps({ name: 'Genre' });
      const e = {
        target: {
          value: 'crime',
        },
      };

      select.props.onChange(e);

      expect(mockCallBack.mock.calls.length).toEqual(1);
      expect(mockCallBack).toHaveBeenCalledWith('crime');
    });
  });
});
