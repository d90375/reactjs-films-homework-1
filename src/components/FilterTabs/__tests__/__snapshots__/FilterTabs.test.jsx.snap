// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTabs tests FilterTabs render FilterTabs renders correctly 1`] = `
<div
  className="container"
>
  <FilterTab
    condition="123"
    fetchByFilter={[MockFunction]}
    filter="Trending"
  >
    Trending
  </FilterTab>
  <FilterTab
    condition="123"
    fetchByFilter={[MockFunction]}
    filter="Top Rated"
  >
    Top Rated
  </FilterTab>
  <FilterTab
    condition="123"
    fetchByFilter={[MockFunction]}
    filter="Coming soon"
  >
    Coming soon
  </FilterTab>
  <select
    className="select active"
    defaultValue="123"
    name="Genre"
    onChange={[Function]}
  >
    <option
      disabled={true}
      hidden={true}
      value="Genre"
    >
      Genre
    </option>
    <option
      value={0}
    >
      drama
    </option>
    <option
      value={1}
    >
      crime
    </option>
    <option
      value={2}
    >
      detective
    </option>
  </select>
</div>
`;

exports[`FilterTabs tests FilterTabs render FilterTabs renders correctly 2`] = `
<div
  className="container"
>
  <FilterTab
    condition="Trending"
    fetchByFilter={[MockFunction]}
    filter="Trending"
  >
    Trending
  </FilterTab>
  <FilterTab
    condition="Trending"
    fetchByFilter={[MockFunction]}
    filter="Top Rated"
  >
    Top Rated
  </FilterTab>
  <FilterTab
    condition="Trending"
    fetchByFilter={[MockFunction]}
    filter="Coming soon"
  >
    Coming soon
  </FilterTab>
  <select
    className="select"
    defaultValue="Genre"
    name="Genre"
    onChange={[Function]}
  >
    <option
      disabled={true}
      hidden={true}
      value="Genre"
    >
      Genre
    </option>
    <option
      value={0}
    >
      drama
    </option>
    <option
      value={1}
    >
      crime
    </option>
    <option
      value={2}
    >
      detective
    </option>
  </select>
</div>
`;
